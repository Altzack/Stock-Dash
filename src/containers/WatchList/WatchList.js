import React, { useContext } from 'react';
import styled from 'styled-components/macro';
import WatchListContainer from '../WatchList/WatchListItem';
import AppContext from '../../AppContext';
import Loader from '../common/Loader/Loader';

const WatchListDiv = styled.div`
  height: calc(100% - 200px);
  overflow-y: auto;
  padding: 15px;
  position: fixed;
  margin-bottom: 10px;
  margin-top: 10px;
  border: 1px solid #fff;
  background-color: rgb(27, 29, 30);
  scrollbar-width: none;
  top: 95px;
  display: block;
  width: 323.77778px;
  @media (min-width: 300px) {
    width: 300px;
    height: calc(100% - 400px);
    position: static;
  }
  @media (min-width: 330px) {
    width: 340px;
    position: static;
  }
  @media (min-width: 600px) {
    width: 550px;
    height: calc(100% - 100px);
  }
  @media (min-width: 900px) {
    width: 800px;
    height: calc(100% - 100px);
  }
  @media (min-width: 1025px) {
    width: 323px;
    position: fixed;
    margin-left: 110px;
    height: calc(100% - 600px);
  }
`;

const SubTitle = styled.div`
  font-family: Rubik;
  color: #e8e6e3;
  font-weight: 400;
  text-align: center;
  margin-top: 5px;
  font-size: 10px;
  @media (min-width: 300px) {
    font-size: 8px;
  }
  @media (min-width: 600px) {
    font-size: 8px;
  }
  @media (min-width: 900px) {
    font-size: 8px;
  }
`;

const PageContainer = styled.div`
  display: flex;
  width: 100%;
  flex-direction: column;
  @media (max-width: 1024px) {
    justify-content: center;
    align-items: center;
  }
`;
const StyledButton = styled.button`
  color: #fff;
  font-weight: 500;
  height: 34px;
  background-color: rgb(40, 199, 145);
  width: 55px;
  outline: none;
  transition: all 0.1s ease-in-out;
  border: 1px solid rgba(0, 0, 0, 0.21);
  border-bottom: 4px solid rgba(0, 0, 0, 0.21);
  border-radius: 4px;
  text-shadow: 0 1px 0 rgba(0, 0, 0, 0.15);
  :hover {
    transition: all 0.1s ease-in-out;
    background-color: rgb(48, 240, 176);
    cursor: pointer;
  }
  @media (max-width: 650px) {
    margin-top: 5px;
  }
  @media (min-width: 655px) {
    margin-top: 10px;
  }
`;

const StyledTitle = styled.h1`
  color: #e8e6e3;
  font-family: Rubik;
  margin-top: 10px;
  @media (min-width: 300px) {
    font-size: 20px;
    letter-spacing: 2px;
  }
  @media (min-width: 650px) {
    font-size: 25px;
    margin-left: 5px;
  }
`;

export default function WatchList() {
  const context = useContext(AppContext);

  return (
    <PageContainer>
      <WatchListDiv>
        <div
          style={{
            display: 'flex',
            justifyContent: 'space-between',
            height: 60,
            borderBottom: '1px solid #fff',
            marginBottom: 10,
          }}
        >
          <StyledTitle>Your top five</StyledTitle>
          {context.editing ? (
            <StyledButton onClick={context.editingOff}>Done</StyledButton>
          ) : (
            <StyledButton onClick={context.setEditing}>Edit</StyledButton>
          )}
        </div>
        {context.loading ? (
          <div style={{ marginTop: 150 }}>
            <Loader />
          </div>
        ) : (
          <>
            <WatchListContainer />
            <SubTitle>click symbol to get news</SubTitle>
          </>
        )}
      </WatchListDiv>
    </PageContainer>
  );
}
