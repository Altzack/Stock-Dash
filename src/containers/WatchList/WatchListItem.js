import React, { useContext } from 'react';
import styled from 'styled-components/macro';
import AppContext from '../../AppContext';
import { RiDeleteBin2Fill } from 'react-icons/ri';
import { message } from 'antd';
import config from '../../config';

const WatchListContainer = styled.div`
  padding: 10px;
  font-family: Rubik;
  margin-bottom: 10px;
  color: #fff;
  font-size: 15px;
  display: flex;
  justify-content: space-evenly;
  width: 100%;
  @media (min-width: 900px) {
    justify-content: space-evenly;
  }
  :hover {
    background-color: rgb(40, 47, 51);
    cursor: pointer;
  }
`;

const WatchListItems = styled.div`
  text-align: left;
`;

function WatchListItem() {
  const context = useContext(AppContext);

  const formatter = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    minimumFractionDigits: 2,
  });

  const listOnSelect = (e) => {
    e.preventDefault();
    const selectedSymbol = e.currentTarget.id;

    fetch(
      `${config.NEWS_API_ENDPOINT}/search?q=${selectedSymbol}&lang=en&sortby=publishedAt&country=us&token=${config.NEWS_API_KEY}`,
      {
        method: 'GET',
        headers: {},
      }
    )
      .then((res) => {
        if (!res.ok) {
          throw new Error(res.status);
        }
        return res.json();
      })
      .then(context.setNews)
      .then(context.setGraphTicker(selectedSymbol))
      .then(context.getGraphData(selectedSymbol))
      .catch((err) => {
        message.error(`Please try again later: ${err}`);
      });

    fetch(
      `${config.DATA_API_ENDPOINT}&symbol=${selectedSymbol}&apikey=${config.DATA_API_KEY}`,
      {
        method: 'GET',
        headers: {},
      }
    )
      .then((res) => {
        if (!res.ok) {
          throw new Error(res.status);
        }
        return res.json();
      })
      .then(context.setGraphTickerPrice)
      .catch((err) => {
        message.error(`Please try again later: ${err}`);
      });
  };

  const onDelete = (e) => {
    e.preventDefault();
    const symbolId = e.currentTarget.id;
    fetch(`${config.API_ENDPOINT}/watchlist/${symbolId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json',
      },
    })
      .then((res) => {
        if (!res.ok) return res.json().then((e) => Promise.reject(e));
      })
      .then(() => {
        context.deleteSymbol(symbolId);
        message.success('Symbol successfully deleted!');
        window.location.reload();
        context.getWatchlist();
      })
      .catch((err) => {
        message.error(`Please try again later: ${err}`);
      });
  };

  return (
    <>
      {context.watchList.map((symbol) => {
        return (
          <div key={symbol.id} id={symbol.symbol}>
            {context.editing ? (
              <WatchListContainer key={symbol.id} id={symbol.symbol}>
                <RiDeleteBin2Fill
                  onClick={onDelete}
                  id={symbol.id}
                  style={{
                    cursor: 'pointer',
                    fontSize: 20,
                    outline: 'none',
                    backgroundColor: 'rgb(27, 29, 30)',
                    border: 'none',
                    padding: 0,
                  }}
                />
                <WatchListItems>{symbol.symbol}</WatchListItems>
              </WatchListContainer>
            ) : (
              <WatchListContainer
                onClick={listOnSelect}
                key={symbol.id}
                id={symbol.symbol}
              >
                <WatchListItems>{symbol.symbol}</WatchListItems>
                {context.closePrice.map((price) => {
                  return price.symbol === symbol.symbol ? (
                    <>
                      <div>
                        <WatchListItems>
                          {price.previousClose > price.price ? (
                            <div style={{ color: 'red' }}>
                              {formatter.format(price.price)}
                            </div>
                          ) : (
                            <div style={{ color: 'rgb(40, 199, 145)' }}>
                              {formatter.format(price.price)}
                            </div>
                          )}
                        </WatchListItems>
                      </div>
                      <div>
                        <WatchListItems>
                          {price.previousClose > price.price ? (
                            <div style={{ color: 'red' }}>
                              {price.change.slice(0, 5)}%
                            </div>
                          ) : (
                            <div style={{ color: 'rgb(40, 199, 145)' }}>
                              +{price.change.slice(0, 4)}%
                            </div>
                          )}
                        </WatchListItems>
                      </div>
                    </>
                  ) : (
                    ''
                  );
                })}
              </WatchListContainer>
            )}
          </div>
        );
      })}
    </>
  );
}
export default WatchListItem;
