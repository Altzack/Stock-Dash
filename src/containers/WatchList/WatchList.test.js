import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WatchList from './WatchList';

it('renders a Watchlist by default', () => {
  const wrapper = shallow(<WatchList />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
