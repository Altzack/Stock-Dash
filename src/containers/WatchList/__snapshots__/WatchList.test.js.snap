// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Watchlist by default 1`] = `
<WatchList__PageContainer>
  <WatchList__WatchListDiv>
    <div
      style={
        Object {
          "borderBottom": "1px solid #fff",
          "display": "flex",
          "height": 60,
          "justifyContent": "space-between",
          "marginBottom": 10,
        }
      }
    >
      <WatchList__StyledTitle>
        Your top 5
      </WatchList__StyledTitle>
      <WatchList__StyledButton
        onClick={[Function]}
      >
        Edit
      </WatchList__StyledButton>
    </div>
    <div
      style={
        Object {
          "marginTop": 150,
        }
      }
    >
      <Loader />
    </div>
  </WatchList__WatchListDiv>
</WatchList__PageContainer>
`;
